src/crimson/osd/osd_operations/pg_splitting.cc

```diff
@@ -0,0 +1,150 @@
+#include "crimson/osd/pg.h"
+#include "crimson/osd/osdmap_service.h"
+#include "crimson/osd/shard_services.h"
+#include "crimson/osd/osd_operations/pg_advance_map.h"
+#include "crimson/osd/osd_operations/pg_splitting.h"
+#include "crimson/osd/osd_operation_external_tracking.h"
+#include "osd/PeeringState.h"
+
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_osd);
+  }
+}
+
+namespace crimson::osd {
+
+PGSplitting::PGSplitting(
+  Ref<PG> pg, ShardServices &shard_services, OSDMapRef cur_map, epoch_t next_epoch,
+  PeeringCtx &&rctx)
+  : pg(pg), shard_services(shard_services), cur_map(cur_map), next_epoch(next_epoch),
+    rctx(std::move(rctx))
+{
+  logger().debug("{}: initiated", *this);
+}
+
+PGSplitting::~PGSplitting() {}
+
+void PGSplitting::print(std::ostream &lhs) const
+{
+  lhs << "PGSplitting("
+      << "pg=" << pg->get_pgid()
+      << " from=" << cur_map->get_epoch()
+      << " to=" << next_epoch;
+  lhs << ")";
+}
+
+void PGSplitting::dump_detail(Formatter *f) const
+{
+    f->open_object_section("PGSplitting");
+    f->dump_stream("pgid") << pg->get_pgid();
+    f->dump_int("from", cur_map->get_epoch());
+    f->dump_int("to", next_epoch);
+    f->close_section();
+}
+
+
+seastar::future<> PGSplitting::start()
+{
+  logger().debug("{}: start", *this);
+  unsigned old_pg_num;
+  using cached_map_t = OSDMapService::cached_map_t;
+
+  if (cur_map->have_pg_pool(pg->get_pgid().pool())) {
+    old_pg_num = cur_map->get_pg_num(pg->get_pgid().pool());
+  } else {
+    logger().debug("{} pool doesn't exist in {}", pg->get_pgid(), cur_map);
+    return seastar::now();
+  }
+  
+  return shard_services.get_map(next_epoch).then(
+    [this, old_pg_num] (cached_map_t&& next_map) {
+    unsigned new_pg_num = next_map->get_pg_num(pg->get_pgid().pool());
+    if (new_pg_num && new_pg_num > old_pg_num) {
+      logger().debug(" New PGs: {} and Old PGs: {}", new_pg_num, old_pg_num);
+      logger().debug(" Checking for PG splitting ");
+      std::set<spg_t> children;
+      if (pg->get_pgid().is_split(old_pg_num, new_pg_num, &children)) {
+        logger().debug(" {} PG has split ", pg->get_pgid());
+        logger().debug(" {} has {} children ", pg->get_pgid(), children.size());
+        std::set<Ref<PG>> split_pgs;
+        return seastar::do_with(pg, children,
+	                        std::move(next_map),
+				new_pg_num,
+				split_pgs, [this] (auto &pg,
+						   auto &children,
+						   auto &next_map,
+						   auto &new_pg_num,
+						   auto &split_pgs) {
+	  return seastar::do_for_each(children, [this, &pg, &next_map, &new_pg_num, &split_pgs] (auto& child_pg) {
+	    return shard_services.get_or_create_pg(child_pg).then([child_pg]
+		 (auto core) {
+		 logger().debug(" PG {} mapped to {}", child_pg.pgid, core);
+		 return seastar::now();
+	       }).then([this, child_pg, next_map=next_map] {
+		 logger().debug(" {} map epoch: {}", child_pg.pgid, next_map->get_epoch());
```

Comment:
Let's use the new logging macros we have.  You can grep for DEBUGDPP in scrub_events/client_requests.